controller/api/destination/endpoint_translator.go

```diff
@@ -494,8 +494,8 @@ func createWeightedAddr(address watcher.Address, opaquePorts map[uint32]struct{}
 	// If the pod is controlled by any Linkerd control plane, then it can be
 	// hinted that this destination knows H2 (and handles our orig-proto
 	// translation)
-	weightedAddr.ProtocolHint = &pb.ProtocolHint{}
 	if controllerNSLabel != "" && !isSkippedInboundPort {
+		weightedAddr.ProtocolHint = &pb.ProtocolHint{}
```

Comment:
This doesn't match my understanding of how this should work.

https://github.com/linkerd/linkerd2-proxy-api/blob/bdcb2d2f5a64d9eb41eaf03216ffce1972f9d787/proto/destination.proto#L130-L145 give us:

```protobuf
    // Hints that the destination will handle this connection as an opaque TCP
    // stream, and (if `opaque_transport` is set) that the proxy should not send
    // a `SessionProtocol` as part of its transport header.
    Opaque opaque = 3;
```
and
```protobuf
  // When set, indicates that the target supports receiving opaque traffic
  // wrapped with the Linkerd connection header on the specified port.
  OpaqueTransport opaque_transport = 2;
```

I think the specific woding is actually exactly backwards; it should read: _and (iff `opaque_transport` is set) that the proxy should send a `SessionProtocol` as part of its transport header._

That is I think we should be setting
```go
                weightedAddr.ProtocolHint.Protocol = &pb.ProtocolHint_Opaque_{
					Opaque: &pb.ProtocolHint_Opaque{},
				}
```
for these unmeshed workloads.

I would expect proxies to fail to honor the opaque port hinting on the outbound side if we don't do this.